cmd/scollector/collectors/collectors.go

```diff
@@ -325,3 +325,17 @@ func skipMetric(index string) bool {
 	}
 	return true
 }
+
+type tsIntegrator func(int64, float64) float64
+
+func getTsIntegrator() tsIntegrator {
+	var i float64
```

Comment:
How about
 ```
type tsIntegrator struct{
   data map[string]struct{
      total floats64
      lastTime
   }
   sync.Mutex
}
```
?

then you can do `cpuIntegrator.Add(key, now, val)` and `cpuIntegrator.Total(key)`. Hide all of the mutexing and map stuff inside the integrator.